app/components/repository-layout.js

```diff
@@ -36,5 +38,15 @@ export default Component.extend({
     toggleTriggerBuildModal() {
       this.toggleProperty('isShowingTriggerBuildModal');
     }
+  },
+
+  didRender() {
+    const repo = this.get('repo');
+
+    if (!repo.canOwnerBuild && repo.isPrivate) {
+      this.flashes.custom('flashes/negative-balance-private', { owner: repo.owner, isUser: repo.ownerType === 'user' });
+    } else if (!repo.canOwnerBuild && !repo.isPrivate) {
+      this.flashes.custom('flashes/negative-balance-public', { owner: repo.owner, isUser: repo.ownerType === 'user' });
```

Comment:
The same goes here with regards to the notifications management. Except that here it may even lead to bugs. Like, when user visits a repository that belongs to the org with negative balance, the message appears, then user navigates to another repository that belongs to another org with a positive balance, but the message is still there, so it confuses the user. Does it make sense?

Reply:
Thank you for noticing this potential behavior. I added cleanup for Flashes.